src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -173,6 +174,36 @@ public function loadClass($class)
                 @trigger_error(sprintf('The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
             }
 
+            // Only check final methods if the class has a parent
+            if ($parent) {
+                foreach ($refl->getMethods(\ReflectionMethod::IS_PUBLIC | \ReflectionMethod::IS_PROTECTED) as $method) {
+                    try {
+                        $prototype = $method->getPrototype();
```

Comment:
@nicolas-grekas we would have to support the edge case I described above and maybe others i didn't think about, we would also have to parse every method doc blocks which is bad for perfs, imo it is better this way.

> This means each class should only look at methods that have $method->class === $class 

It looks like I forgot this check...